### ISSUE TITLE
TODO: octoshock update testing

### ISSUE THREAD

**Issue by zeromus** [TASVideos/BizHawk#1912]
- [ ] test all games mentioned in mednafen changelog
- [ ] test screen shakes
- [ ] test a multi-disc game
- [ ] test any guns that worked before (incl. mouse)

**Comment by Spikestuff** [comment]
Just as a quick note screen scaling/positioning is currently an issue.
Tested Game: Crash Bandicoot (USA) [560x240] -- Including Crash Warped and Crash Team Racing
- Pixel Pro Mode: Correct.
- Both Mednafen Modes: roughly 24 pixels from that top can be seen on screen.
- Hardcore Debug is wider than what it normally is.
- Dumping/Screenshots is also wider than what it should be.
(800x240 not 560x240 on Screenshots/Dumping/Hardcore Debug)

350x240 resolution games such as Incredible Crisis and Resident Evil Survivor have the same issues as above, only in Mednafen Mode more of the screen is visible (168 pixels from the top).

[This is using the latest commit](https://github.com/TASVideos/BizHawk/commit/df538c2833152ec0fe633d4c0eccb3b9af827d60) off of appveyor.

**Comment by zeromus** [comment]
I just did a bunch of work to fix this kind of stuff.

I didn't trace through your scenarios but rather just broadly investigated a bunch of screwups.
Can you check it again? If you find a single scenario that's broken, just post the one rather than proceed to an exhaustive test suite.

**Comment by zeromus** [comment]
Also feel free to edit my OP (if you can) or post which of those items you feel are tested.
I suspect the light guns are all messed up. Is there a game or homebrew with a calibration mode for testing?

**Comment by Spikestuff** [comment]
> Is there a game or homebrew with a calibration mode for testing?

All Namco titles that have gun support a calibration mode:
- Point Blank 1, 2 & 3
- Time Crisis 1 & 2

I'll get onto checking it and asking someone else to check for GunCons cause I know they've been looking forward to that feature.

> Also feel free to edit my OP (if you can) 

Something I can't do unfortunately since I'm not a contributor.
---
There's an issue that I don't know if it could be adjusted or not, but I want to point it out.
The current DevBuilds of 2.5 is faster in loading compared to 2.4 and compared to an actual 5501 console.

Test case is [Crash Bandicoot](https://youtu.be/QWD86emV67Q) since I have the most footage on that title.
The short version is to finish all those demos is (ms might be slightly off).
They're all using SCPH5501 except for the PS2s just to show outliers of the PS2 version of loadings and what was gathered there (XEBRA is also shown as an outlier for slowest loads):
- XEBRA - 11:30.12
- _**PS1: 11:27.22**_
- 2.4: 11:25.39
- Dev 2.5: 11:20.12
- PS2 - 11:20.25 (75001 & Slow Disk Enabled)
- PS2 - 11:18.37 (90001 & Slow Disk Enabled)
- mednafen: 11:14.19

**Comment by zeromus** [comment]
Uhh if guncons never worked before, then they certainly won't now. 
About timing:
1. That's an issue to file with mednafen, not bizhawk
2. But don't file that issue with mednafen, mednafen's timing is tuned for internal compatibility, not a perfect reflection of reality. 
3. File an issue with bizhawk if its timing is different than the version of mednafen it tracks, which is 1.24.0-UNSTABLE. But count FRAMES when testing. Mednafen may write different values for its QT dump timing than bizhawk does. If bizhawk is truly as different from mednafen as you say, then something is terribly wrong.

**Comment by ThunderAxe31** [comment]
Emulation for Rescue Shot Bubibo has regressed a huge lot in the latest devbuild.
See these screenshots: https://imgur.com/a/q5S2f7v
The former 4 screenshots are from BizHawk 2.4, while the latter 4 ones are from BizHawk_Developer-2020-04-09-004904-#4f0849681586ed38f0bea46374f1522e2d994719

game .bin file checksums (matches No-Intro):
```
    File: Rescue Shot Bubibo (Japan).bin
  CRC-32: ec44c20c
     MD5: 102d7580d050b12aacc09ebab6ea1047
   SHA-1: 27e921b97ddb2f1703f8d16a4a40399857392860
 SHA-256: 65979175ccab07f729c591882c28e07037140812e96de30252a7c43eba90cbf0
```
game .cue file contents:
```
FILE "Rescue Shot Bubibo (Japan).bin" BINARY
  TRACK 01 MODE2/2352
    INDEX 01 00:00:00
```

**Comment by zeromus** [comment]
fixed

**Comment by YoshiRulz** [comment]
Should also check #1686 (exception when using Debugger tool wasn't handled correctly).

**Comment by vadosnaprimer** [comment]
Related to https://github.com/TASVideos/BizHawk/issues/1225

**Comment by ThunderAxe31** [comment]
Is this the kind of shakes you needed to be tested?
https://gfycat.com/elasticexcellenteland
https://gfycat.com/neardelectabledotterel
These were recording with the latest devbuild: BizHawk_Developer-2020-04-28-012651-#c8ae659b1994a0759fb85ef3bea0b6b6e18add85
Note that, technically speaking, in these recording it's actually the in-game camera that gets moved. In fact, you can see that close objects move more than far objects.

**Comment by zeromus** [comment]
No, I mean horizontal framebuffer shakes, as in chrono cross I believe

**Comment by vadosnaprimer** [comment]
Seems to be ok? It's been released anyway.
